libraries/standard/serializer/test/unit/iot_tests_serializer_cbor.c

```diff
@@ -445,6 +453,201 @@ static const uint8_t _testEncodedNestedMap[] =
+            0x65, 0x64,
+        0xFF,                                                                   /* # primitive(*) */
+
+        0x67,                                                                   /* # text(7) */
+        0x73, 0x75, 0x63, 0x63, 0x65, 0x73, 0x73,                               /* # "success" */
+        0x6E,                                                                   /* # text(14) */
+        0x4E, 0x6F, 0x74, 0x20, 0x73, 0x75, 0x63, 0x63, 0x65, 0x73, 0x73, 0x66, /* # "Not successful" */
+        0x75, 0x6C,
+    0xFF,                                                                       /* # primitive(*) */
+};
+
+/* JSON Representation:
+ * {
+ *    "result": {
+ *                 "statusCode": 400,
+ *                 "errorCode": "InvalidPayload",
+ *                 "errorMessage": "Message cannot be parsed"
+ *              }
+ * }
+ */
+static const uint8_t _testEncodedIndefiniteLengthNestedMapWith2BreakChars[] =
+{
+    0xBF,                                                                       /* # map(*) */
+        0x66,                                                                   /* # text(6) */
+        0x72, 0x65, 0x73, 0x75, 0x6C, 0x74,                                     /* # "result" */
+        0xBF,                                                                   /* # map(*) */
+            0x6A,                                                               /* # text(10) */
+            0x73, 0x74, 0x61, 0x74, 0x75, 0x73, 0x43, 0x6F, 0x64, 0x65,         /* # "statusCode" */
+            0x19, 0x01, 0x90,                                                   /* # unsigned(400) */
+
+            0x69,                                                               /* # text(9) */
+            0x65, 0x72, 0x72, 0x6F, 0x72, 0x43, 0x6F, 0x64, 0x65,               /* # "errorCode" */
+            0x6E,                                                               /* # text(14) */
+            0x49, 0x6E, 0x76, 0x61, 0x6C, 0x69, 0x64, 0x50, 0x61, 0x79, 0x6C,   /* # "InvalidPayload" */
+            0x6F, 0x61, 0x64,
+
+            0x6C,                                                               /* # text(12) */
+            0x65, 0x72, 0x72, 0x6F, 0x72, 0x4D, 0x65, 0x73, 0x73, 0x61, 0x67,   /* # "errorMessage"*/
+            0x65,
+            0x78, 0x18,                                                         /* # text(24) */
+            0x4D, 0x65, 0x73, 0x73, 0x61, 0x67, 0x65, 0x20, 0x63, 0x61, 0x6E,   /* # "Message cannot be parsed" */
+            0x6E, 0x6F, 0x74, 0x20, 0x62, 0x65, 0x20, 0x70, 0x61, 0x72, 0x73,
+            0x65, 0x64,
+        0xFF,                                                                   /* # primitive(*) */
+    0xFF,                                                                       /* # primitive(*) */
+};
+
+/* JSON Representation:
+ * {
+ *     "result": h'AABBCCFF'
+ * }
+ */
+static const uint8_t _testEncodedIndefiniteLengthMapWithFFatTheEnd[] =
```

Comment:
Should similar corner cases be tested for arrays as well to be comprehensive in our test coverage?

Reply:
Done.